Automattic/woocommerce-payments
# Add idempotency key to post requests

Fixes #530 

#### Changes proposed in this Pull Request

This adds an idempotency key to all post requests sent to the server. This also adds request headers to logs so the idempotency key is useful for debugging.






#### Testing instructions

- Trigger a post request by doing something like completing checkout
- Confirm idempotency key is present in headers. Be sure to also check logs.
- Trigger a get or delete request
- Confirm idempotency key is not present in headers. Be sure to also check logs.

-------------------


**Post merge**
